### PR TITLE
Show number of commits a branch is ahead or behind the default branch

### LINKED ISSUE #5326: Show number of commits a branch is ahead or behind default
When selecting 'branches' in the UI, it would be helpful when the number of commits a particular branch is ahead or behind the default branch could be displayed (like it can be seen in GitHub or GitLab).

### PR DESCRIPTION
**Pull request by adelowo**
Fixes https://github.com/go-gitea/gitea/issues/5326

This uses a flash message to display "if ahead, behind or equal with the default branch". 

screenshot 2018-11-16 at 00 14 59

### CODE REVIEW

**Comment by adelowo** [comment]
We obviously can improve it by adding it to the UI 
but frontend is not my strength and will appreciate any help if someone has a prototype
of what it is supposed to look like

File: routers/repo/view.go [hunk 1]
```diff
@@ -372,6 +372,56 @@ func renderCode(ctx *context.Context) {
...
+		if err != nil {
+			ctx.ServerError("CommitsCount", err)
+			return
+		}
+
+		n := commitCountInMaster - ctx.Repo.CommitsCount
+
+		var msg string
+
+		if n == 0 {
```
Comment (lunny) [inline routers/repo/view.go]: how about to use a switch?

Comment (lafriks) [reply routers/repo/view.go]: Or better use functions like in activity page that supports translations based on count

**Comment by techknowlogick** [comment]
@adelowo are you able to provide screenshots?

**Comment by adelowo** [comment]
Sure @techknowlogick .. I updated the initial post

**Comment by adelowo** [comment]
___Unstale___

**Comment by adelowo** [comment]
https://github.com/go-gitea/gitea/pull/6695 has a better fix by putting that into a reusable function.

Once it is merged, will update this PR :)

**Comment by lunny** [comment]
image
I think maybe we should add both behind and before?

**Comment by adelowo** [comment]
Can this be reviewed again?

File: routers/repo/view.go [hunk 2]
```diff
@@ -518,6 +519,40 @@ func renderCode(ctx *context.Context) {
...
+
+		if divergence.Ahead > 0 {
+			if divergence.Ahead == 1 {
+				msg = ctx.Tr("repo.commits.count.ahead_1", ctx.Repo.Repository.DefaultBranch)
+			} else {
+				msg = ctx.Tr("repo.commits.count.ahead_n", divergence.Ahead, ctx.Repo.Repository.DefaultBranch)
+			}
+		}
+
+		if divergence.Behind > 0 {
```
Comment (lunny) [inline routers/repo/view.go:704]: if both `divergence.Ahead` and `divergence.Behind` > 0?

Comment (adelowo) [reply routers/repo/view.go:704]: Not sure we need this

**Pull request closed without merging**